interpreter/interpreter.go

```diff
@@ -496,6 +497,21 @@ func (itrp *Interpreter) doStringPart(ctx context.Context, part semantic.StringE
 			return nil, errors.Newf(codes.Invalid, "%s: interpolated expression produced a null value",
 				p.Location())
 		} else if v.Type().Nature() != semantic.String {
+			val := values.Unwrap(v)
+			switch v.Type().Nature() {
+			case semantic.Bool:
+				return values.NewString(strconv.FormatBool(val.(bool))), nil
+			case semantic.Int:
+				return values.NewString(strconv.FormatInt(val.(int64), 10)), nil
+			case semantic.UInt:
+				return values.NewString(strconv.FormatUint(val.(uint64), 10)), nil
+			case semantic.Float:
+				return values.NewString(strconv.FormatFloat(val.(float64), 'f', -1, 64)), nil
+			case semantic.Time:
+				return values.NewString(val.(values.Time).String()), nil
+			case semantic.Duration:
+				return values.NewString(val.(values.Duration).String()), nil
+			}
```

Comment:
Where is the equivalent logic for the compiler? Maybe they can share the same implementation? This logic only uses types from the `values` package, perhaps a helper method in the values package like `Stringify(v Value) (String, error)` would be useful?